File: StocksProcessing.Preprocessor/Models/Messages/CurrencyRateChangeMessage.cs
using System.Text.Json.Serialization;
using StocksProcessing.Generator.Models.Enums;

namespace StocksProcessing.Preprocessor.Models.Messages;

public sealed record CurrencyRateChangeMessage
{
    public required string StockExchangeName { get; init; }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public required Currency Currency { get; init; }

    public required decimal RateInRubles { get; init; }
}

File: StocksProcessing.Analyzer/src/main/java/org/example/models/messages/CurrencyRateChangeMessage.java
package org.example.models.messages;

import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
import org.example.models.Currency;

public class CurrencyRateChangeMessage {
    @JsonProperty("stock_exchange_name")
    public String stockExchangeName;

    @JsonProperty("currency")
    public Currency currency;

    @JsonProperty("rate_in_rubles")
    public Double rateInRubles;
}
